src/services/safe_border.py

```diff
@@ -0,0 +1,134 @@
+from src.web3_extentions.typings import Web3
+from src.typings import BlockStamp
+from src.web3_extentions.lido_validators import LidoValidator
+
+NEW_REQUESTS_BORDER = 8 # epochs ~50 min
+MAX_NEGATIVE_REBASE_BORDER = 1536 # epochs ~6.8 days
+FAR_FUTURE_EPOCH = 2**64 - 1
+MIN_VALIDATOR_WITHDRAWABILITY_DELAY = 2**8
+EPOCHS_PER_SLASHINGS_VECTOR = 2**13
+SLOTS_PER_EPOCH = 2**5
+
+class SafeBorder:
+    def __init__(self, w3: Web3) -> None:
+        self.w3 = w3
+        self.lido_contracts = w3.lido_contracts
+
+    def get_safe_border_epoch(self, is_bunker: bool, blockstamp: BlockStamp):
+        if is_bunker is False:
+            return self.get_new_requests_border_epoch(blockstamp)
+        
+        negative_rebase_border_epoch = self.get_negative_rebase_border_epoch(blockstamp)
+        associated_slashings_border_epoch = self.get_associated_slashings_border_epoch(blockstamp)
+
+        return min(
+            negative_rebase_border_epoch,
+            associated_slashings_border_epoch
+        )
+
+    def get_new_requests_border_epoch(self, blockstamp: BlockStamp):
+        return get_epoch_by_slot(blockstamp.slot_number) - NEW_REQUESTS_BORDER
+
+    def get_negative_rebase_border_epoch(self, blockstamp: BlockStamp):  
+        bunker_start_timestamp = self.get_bunker_mode_start_timestamp(blockstamp)
```

Comment:
need to handle a scenario when the bunker was disabled before (for the first report when is_bunker: true). `bunkerModeSinceTimestamp()` will return maxuint256 in this case

So we need to take the timestamp of the last **successful** report before the current one